### ISSUE TITLE
Lost Thetis functions due to Apache Labs Hardware Changes

### ISSUE THREAD

**Issue by chipp1** [ramdor/Thetis-2.9.0#76]
Newer 7000 MKII radios do not accurately display current draw and voltage. This occurred when Apache Labs made component changes due to part obsolesce or supply chain issues and did not adequately test the hardware to ensure compatibility with Thetis and/or Protocol 2 firmware.

**Comment by ramdor** [comment]
yes it would be nice for them to send us few developers loaner radios or a developer kit to get this stuff ironed out and tested, but I think they forget we exist sadly. I don't mind signing an agreement that will let them have it all back if they collect it.

**Comment by ramdor** [comment]
Addtional config has been added into setup for current draw. The same will be added for voltage at some point.

**Comment by chipp1** [comment]
Thanks Ritchie, wishing you a very enjoyable summer. And many thanks again for all the effort you put into Thetis. It enhances every Anan owner’s enjoyment of the rig.

Best,
Chip – W6MY

**Comment by RdWing** [comment]
This still does not resolve the broken current meter on all 7000's Mk 2.5 and later.

**Comment by ramdor** [comment]
@RdWing I am adding an option to log some data to VALog.txt which users can sent in so that we can try and get to the bottom of the issue.

**Comment by RdWing** [comment]
I collected logs from a few systems...
From a 7000 DLE Mk 2 with the original FPGA, made right before the changeover to the EP4CE Mk 2.5.
[mk2_VALog.txt](https://github.com/ramdor/Thetis/files/12886739/mk2_VALog.txt)

From a 7000 DLE Mk2.5 with the EP4CE FPGA
[VALog.txt](https://github.com/ramdor/Thetis/files/12886743/VALog.txt)

I tried playing with the current sensitivity and offset. If I set the offset to 1.0 it will show 0.2A always but that number doesn't seem to change.

**Comment by ramdor** [comment]
ok, I was going to make a post about this, but haven't had a chance yet. You need to leave it run whilst using an external volt and ammeter. You need to record the times at which you are pulling  constant 5A, 10A, 15A, for example, and the idle current. Also include the voltages at the back of the radio (ignore vdrop). Then send that list of times, A, V, together with the log file. Then I can look at the actual amps being pulled vs what is being reported in Thetis in the log. Of course the radio model as well.

R.

**Comment by RdWing** [comment]
It's going to take me awhile to collect everything needed to do that test, and to actually do that test. If someone else has the necessary hardware please feel free to contribute.

**Comment by ramdor** [comment]
other way is just to set up two meters, volt/ammeter close to back of radio, film it with clock hr/min/seconds. Then i can check times in the log file against the video to see what the V/A was at the log entry time.
